### DISCUSSION TITLE
re macros conflict with macros within the iOS SDK

### DISCUSSION THREAD

**Discussion by MitchellFreeman** [baresip/re#285] [General]
Two macros (that I am aware of), `LIST_INIT` and `LIST_FOREACH` conflict with macros within `sys/queue.h` in the iOS SDK. re can't be built with these due to multiple definitions and they need to be altered in some way before they can build.

**Comment by sreimers** [comment]
Is `sys/queue.h` included by a system library or something you include in your project? Can you share a code/project example?

**Comment by sreimers** [comment]
We simply `undef` these macros before include re headers if needed, like here:

https://github.com/baresip/re/blob/198fbd978b84822101e53d00490fe6bbec57af7e/src/main/main.c#L30-L44

**Comment by sreimers** [comment]
I have documented this a little better here: https://github.com/baresip/re/wiki/Development#conflicting-macro-definitions

I think it is better to solve this in the application itself, because it also makes you aware that there are conflicts. Let me know if this fixes your problems.
